index.js

```diff
@@ -1,48 +1,46 @@
+const defaultsShape = {
+  existingCollaboratorMessage: 'string',
+  newCollaboratorMessage: 'string',
+};
+
+function checkForDefaults() {
+  const errors = Object.keys(defaultsShape).filter(key => !Object.prototype.hasOwnProperty.call(defaults, key));
+  if (errors.length > 0) errors.forEach(err => console.error(`Key \`${err}\` of type \`${defaultsShape[err]}\` is missing.`));
+}
+
 /**
   * @typedef {Object} Config
   * @prop {string} existingCollaboratorMessage
   * @prop {string} newCollaboratorMessage
   *
   * Anytime a user opens an issue, add them as a collaborator to the repository.
   * @param {Object} robot
-  * @param {Config} [defaults]
+  * @param {Config} defaults
   * @param {string} [configFilename]
   */
+module.exports = (robot, defaults, configFilename = 'add-collabs.yml') => {
+  checkForDefaults(defaults);
 
-module.exports = (robot, defaults = {}, configFilename = 'add-collabs.yml') => {
-    let config;
-
-    defaults = Object.assign({}, {
-      addCollaborators: {
-        newCollaboratorMessage: 'Hi! I\'m the friendly :robot: of this repo.\n\nWe\'re happy you\'re here :wave:. Everyone is welcome here, so I\'m making you a collaborator. This will give you access to commit on this repo.',
-        existingCollaboratorMessage: 'Hi! I\'m the friendly :robot: of this repo.\n\nI can see that you\'re already a collaborator. Any other issues are above my paygrade at the moment, so we\'ll have to wait for a pesky hu-man. Not to worry though, they\'ll drop by within 24 hours to answer your questions!'
-      }
-    }, defaults || {});
-
-    robot.on('issues', async context => {
-
+  robot.on('issues.opened', async context => {
     const {issue, action} = context.payload;
     const issueOwner = issue.user.login;
+
     const repo = context.repo({username: issueOwner});
 
-    try {
-      config = await context.config( configFilename );
-    } catch (err) {
-      config = defaults;
-    }
+    const repoConfig = await context.config(configFilename);
```

Comment:
@hectorsector I can confirm that Probot [throws an error](https://github.com/probot/probot/blob/master/test/context.js#L108) when the file doesn't exist. That means that if we wanted to use that try/catch it would make sense. I'll readd that 👍